gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/source/HiveSource.java

```diff
@@ -330,8 +331,10 @@ else if (partition.getTPartition().isSetParameters()
   @Override
   public Extractor getExtractor(WorkUnitState state) throws IOException {
     try {
-      return new HiveConvertExtractor(state, getSourceFs());
-    } catch (TException | HiveException e) {
+      return (Extractor) Class.forName(state.getProp(HIVE_SOURCE_EXTRACTOR_TYPE, DEFAULT_HIVE_SOURCE_EXTRACTOR_TYPE))
```

Comment:
This seems like a good feature for all gobblin sources. Should this be moved to `gobblin.runtime.Task` instead?